platform/ext/target/nordic_nrf/common/core/target_cfg.c

```diff
@@ -1289,15 +1289,48 @@ static const uint8_t target_peripherals[] = {
 	dppi_channel_configuration();
 
     /* GPIO pin configuration */
-#ifdef NRF_SPU
-
-	nrf_spu_gpio_config_set(NRF_SPU, 0, TFM_PERIPHERAL_GPIO0_PIN_MASK_SECURE, SPU_LOCK_CONF_LOCKED);
+	uint32_t secure_pins[] = {
+		TFM_PERIPHERAL_GPIO0_PIN_MASK_SECURE,
 #ifdef TFM_PERIPHERAL_GPIO1_PIN_MASK_SECURE
-	nrf_spu_gpio_config_set(NRF_SPU, 1, TFM_PERIPHERAL_GPIO1_PIN_MASK_SECURE, SPU_LOCK_CONF_LOCKED);
+		TFM_PERIPHERAL_GPIO1_PIN_MASK_SECURE,
 #endif
+#ifdef TFM_PERIPHERAL_GPIO2_PIN_MASK_SECURE
+		TFM_PERIPHERAL_GPIO2_PIN_MASK_SECURE,
+#endif
+	};
+
+	/* Note that there are two different API's for SPU configuration */
+#if NRF_SPU_HAS_MEMORY
+
+	for(int port = 0; port < ARRAY_SIZE(secure_pins); port++){
+		nrf_spu_gpio_config_set(NRF_SPU, port, secure_pins[port], SPU_LOCK_CONF_LOCKED);
+	}
 
+#elif NRF_SPU_HAS_PERIPHERAL_ACCESS
+
+	for(int port = 0; port < 2; port++) {
```

Comment:
Is hard coding this to 2 right, or should it be ARRAY_SIZE(secure_pins) like in the previous case?

Reply:
I can do ARRAY_SIZE :)